web/src/compose_paste.ts

```diff
@@ -38,10 +38,25 @@ function image_to_zulip_markdown(
         // For Zulip's custom emoji
         return node.getAttribute("alt") ?? "";
     }
-    const src = node.getAttribute("src") ?? node.getAttribute("href") ?? "";
-    const title = deduplicate_newlines(node.getAttribute("title") ?? "");
-    // Using Zulip's link like syntax for images
-    return src ? "[" + title + "](" + src + ")" : (node.getAttribute("alt") ?? "");
+    // For inline images presented in Markdown syntax, we need
+    // to grab the `data-original-src` attribute, rather than
+    // the thumbnailed image referenced on the `src` attribute
+    const src =
+        // eslint-disable-next-line unicorn/prefer-dom-node-dataset
+        node.getAttribute("data-original-src") ??
+        node.getAttribute("src") ??
+        node.getAttribute("href") ??
+        "";
+    // We prioritize the alt attribute, which will be present on
+    // inline Markdown images
```

Comment:
What's the motivation for preferring `title` vs `alt`? It sounds like the corner case you're thinking about is one where only `alt` is present.

Reply:
``` diff
diff --git a/web/src/compose_paste.ts b/web/src/compose_paste.ts
index 880fef70cf..f8fb9b4201 100644
--- a/web/src/compose_paste.ts
+++ b/web/src/compose_paste.ts
@@ -48,8 +48,10 @@ function image_to_zulip_markdown(
         node.getAttribute("src") ??
         node.getAttribute("href") ??
         "";
-    // We prioritize the alt attribute, which will be present on
-    // inline Markdown images
+    // For the label, we prioritize the alt attribute, which will be
+    // present on both Zulip and third-party Markdown images, but use
+    // title as a fallback, since older-style Zulip image previews
+    // only set title.
     let alt = "";
     if (node.classList.contains("inline-image")) {
         alt = node.getAttribute("alt") ?? "";
```